app/models/sac_cas/role.rb

```diff
@@ -8,23 +8,73 @@
 module SacCas::Role
   module ClassMethods
     def select_with_membership_years(date = Time.zone.today)
+      # because parameter passed in query is CET we make sure to convert all database dates from UTC to CET
       <<~SQL
-        CASE
-          -- membership_years is only calculated for Mitglied roles
-          WHEN roles.type != "Group::SektionsMitglieder::Mitglied" THEN 0
-          ELSE (
-            1 + DATEDIFF(
-                  LEAST(
-                    -- LEAST will return NULL if any of the arguments is NULL.
-                    -- Any value that can be NULL must have a fallback value higher than date.
-                    '#{date.strftime("%Y-%m-%d")}',
-                    COALESCE(DATE(roles.deleted_at), '9999-12-31'),
-                    COALESCE(DATE(roles.archived_at), '9999-12-31'),
-                    COALESCE(roles.delete_on, '9999-12-31')
-                  ),
-                  DATE(roles.created_at)
-                )
-            )/365
+              CASE
+                -- membership_years is only calculated for Mitglied roles
+                WHEN roles.type != 'Group::SektionsMitglieder::Mitglied' THEN 0
+                ELSE (
+                    TIMESTAMPDIFF(
+                        YEAR, 
+                        DATE(CONVERT_TZ(roles.created_at, "UTC", "CET")),
+                        LEAST(
+                            DATE('#{date.strftime("%Y-%m-%d")}'),
+                            -- add one day to include last day in calculation
+                            COALESCE(DATE(CONVERT_TZ(roles.deleted_at, "UTC", "CET")) + INTERVAL 1 DAY, '9999-12-31'),
+                            COALESCE(DATE(CONVERT_TZ(roles.archived_at, "UTC", "CET")) + INTERVAL 1 DAY, '9999-12-31'),
+                            COALESCE(roles.delete_on + INTERVAL 1 DAY, '9999-12-31')
+                        )
```

Comment:
Dieser Block wiederholt sich und könnte in eine eigene Methode extrahiert werden